.github/workflows/get-dependency-details.yml

```diff
@@ -0,0 +1,108 @@
+      - name: Get dependency details
+        id: dependency
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          echo "Parsing JSON payload: '${{ inputs.payload }}'..."
+          ID=""
+
+          if ! [[ -n "${{ inputs.payload }}" && $(echo '${{ inputs.payload }}' | jq 'has("${{ inputs.repo }}")') == true ]]; then
+              # Repo not specified in payload; check version.json
+              if [ "${{ inputs.skip-version-json-check }}" = true ]; then
+                echo "No JSON payload given with repo ${{ inputs.repo }}, and skip-version-json-check flag is set. Exiting."
+                exit 0
+              fi
+              echo "No JSON payload given with repo ${{ inputs.repo }}, using version.json"
+              ID=$(jq -r '.dependencies."${{ inputs.repo }}"' version.json)
+              echo "Found build ID for ${{  inputs.repo }} in version.json: '$ID'"
+          else
+              ID=$(echo '${{ inputs.payload }}' | jq -r '."${{ inputs.repo }}"')
+              echo "Payload contains build ID for ${{ inputs.repo }}: '$ID'"
+          fi
+
+          echo "Searching for ${{ inputs.repo }} build id '$ID'..."
+          if gh release view --repo ${{ inputs.owner }}/${{ inputs.repo }} "$ID" > /dev/null 2>&1; then
+              # Release
+              echo "Found release build"
+              echo "run-type=$(echo "release")" >> $GITHUB_OUTPUT
+              # Set metadata (sha, url)
+              head_sha=$(gh release view --repo ${{ inputs.owner }}/${{ inputs.repo }} $ID --json targetCommitish | jq '.targetCommitish')
+              url=$(gh release view --repo ${{ inputs.owner }}/${{ inputs.repo }} $ID --json url | jq '.url')
+              echo "head-sha=$(echo "$head_sha")" >> $GITHUB_OUTPUT
+              echo "url=$(echo "$url")" >> $GITHUB_OUTPUT
+          elif gh run view --repo ${{ inputs.owner }}/${{ inputs.repo }} "$ID" > /dev/null 2>&1; then
+              # Prerelease (workflow run)
+              echo "Found non-release build"
+              echo "run-type=$(echo "run")" >> $GITHUB_OUTPUT
+              # Set metadata (build date, title, sha, url)
+              build_date=$(gh run view --repo ${{ inputs.owner }}/${{ inputs.repo }} $ID --json createdAt | jq '.createdAt')
+              build_title=$(gh run view --repo ${{ inputs.owner }}/${{ inputs.repo }} $ID --json displayTitle | jq '.displayTitle')
+              head_sha=$(gh run view --repo ${{ inputs.owner }}/${{ inputs.repo }} $ID --json headSha | jq '.headSha')
+              url=$(gh run view --repo ${{ inputs.owner }}/${{ inputs.repo }} $ID --json url | jq '.url')
+              echo "build-date=$(echo "$build_date")" >> $GITHUB_OUTPUT
+              echo "build-title=$(echo "$build_title")" >> $GITHUB_OUTPUT
+              echo "head-sha=$(echo "$head_sha")" >> $GITHUB_OUTPUT
+              echo "url=$(echo "$url")" >> $GITHUB_OUTPUT
+          else
+              echo "Build not found"
+              exit 1
+          fi
```

Comment:
This section calls out to the same base API a bunch of times. I would lift that up to only call once, store response, and parse from that instead of continually calling gh run/release.